ProcessLib/Parameter/Parameter.h

```diff
@@ -54,6 +54,24 @@ struct Parameter : public ParameterBase
     //! Returns the parameter value at the given time and position.
     virtual std::vector<T> const& getTuple(
         double const t, SpatialPosition const& pos) const = 0;
+
+    //! Scalar value parameter access. The number of components of this
+    //! parameter must be 1.
+    double operator()(double const t, SpatialPosition const& x) const
+    {
+        assert(getNumberOfComponents() == 1);
+        return getTuple(t, x)[0];
```

Comment:
`front` and `operator[](0)` result in same binary code (at least when optimized).
When not optimized `operator[](int n)` is implemented roughly as `return *(this->data + n)`, whereas `front()` is `return *begin()` = `return *(iterator(this->data))`, so it creates a temporary iterator there... (At least in the gcc-6.1.1 implementation of the STL)